### DISCUSSION TITLE
Lesson 3 doesn't compile

### DISCUSSION THREAD

**Discussion by macinsmith** [pointfreeco/episode-code-samples#149] [Q&A]
Lesson3 has several issues

1. baseButtonStyle - UIEdgeInsets is deprecated
2. borderButtonStyle - UIView has no member setTitleColor
3. Canot find GradientView in scope

I attempted to open the PointFree.playground at the root of the repo and it could not find HttpPipeline to import it.

**Comment by mbrandonw** [comment]
Hi @macinsmith, that episode is now 5 years old and a lot has changed in Swift since then. It's not really possible for us to keep all of the sample codes up-to-date, and even if we did, then it wouldn't represent what we covered in the episode.
